cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java

```diff
@@ -779,6 +781,72 @@ public void testBuildEmptyPackageWithCompilerPlugin() throws IOException {
                 getOutput("build-empty-package.txt"));
     }
 
+    @Test(description = "Build a ballerina project with the flag dump-graph")
+    public void testBuildBalProjectWithDumpGraphFlag() throws IOException {
+        Path dumpGraphResourcePath = this.testResources.resolve("projectsForDumpGraph");
+        BCompileUtil.compileAndCacheBala(dumpGraphResourcePath.resolve("package_c"), testDistCacheDirectory,
+                projectEnvironmentBuilder);
+        BCompileUtil.compileAndCacheBala(dumpGraphResourcePath.resolve("package_b"), testDistCacheDirectory,
+                projectEnvironmentBuilder);
+
+        Path projectPath = dumpGraphResourcePath.resolve("package_a");
+        System.setProperty("user.dir", projectPath.toString());
+
+        java.io.ByteArrayOutputStream out = new java.io.ByteArrayOutputStream();
+        System.setOut(new java.io.PrintStream(out));
+
+        BuildCommand buildCommand = new BuildCommand(projectPath, printStream, printStream, false);
+        new CommandLine(buildCommand).parseArgs("--dump-graph");
+        buildCommand.execute();
+        String buildLog = readOutput(true).replaceAll("\r", "");
+        String dependencyGraphLog = out.toString().replaceAll("\r", "");
+
+        Assert.assertTrue(dependencyGraphLog.contains(getOutput("dump-graph-output.txt")));
```

Comment:
Here in `dump-graph-output.txt` we have full build output or part of the build output?
If it's the full output we can use `assertEquals` instead of `assetTrue`, So when this assertion fails diff will print in the console.
WDYT?

Reply:
Right now `dump-graph-output.txt` has only a part of the output since the same graph is outputted 3/4 times in the console. I need to find why that is happening exactly. The current o/p of --dump-graph for package A is,
```
Resolving dependencies
digraph "foo/package_a:0.1.0" {
	node [shape=record]
	"foo/package_a" [label="<0.1.0> foo/package_a:0.1.0"];
	"foo/package_b" [label="<0.1.0> foo/package_b:0.1.0"];
	"foo/package_c" [label="<0.1.0> foo/package_c:0.1.0"];

	// Edges
	"foo/package_a":"0.1.0" -> "foo/package_b":"0.1.0";
	"foo/package_b":"0.1.0" -> "foo/package_c":"0.1.0";
}

Resolving dependencies
digraph "foo/package_a:0.1.0" {
	node [shape=record]
	"foo/package_a" [label="<0.1.0> foo/package_a:0.1.0"];
	"foo/package_b" [label="<0.1.0> foo/package_b:0.1.0"];
	"foo/package_c" [label="<0.1.0> foo/package_c:0.1.0"];

	// Edges
	"foo/package_a":"0.1.0" -> "foo/package_b":"0.1.0";
	"foo/package_b":"0.1.0" -> "foo/package_c":"0.1.0";
}

Resolving dependencies
digraph "foo/package_a:0.1.0" {
	node [shape=record]
	"foo/package_a" [label="<0.1.0> foo/package_a:0.1.0"];
	"foo/package_b" [label="<0.1.0> foo/package_b:0.1.0"];
	"foo/package_c" [label="<0.1.0> foo/package_c:0.1.0"];

	// Edges
	"foo/package_a":"0.1.0" -> "foo/package_b":"0.1.0";
	"foo/package_b":"0.1.0" -> "foo/package_c":"0.1.0";
}

Resolving dependencies
digraph "foo/package_a:0.1.0" {
	node [shape=record]
	"foo/package_a" [label="<0.1.0> foo/package_a:0.1.0"];
	"foo/package_b" [label="<0.1.0> foo/package_b:0.1.0"];
	"foo/package_c" [label="<0.1.0> foo/package_c:0.1.0"];

	// Edges
	"foo/package_a":"0.1.0" -> "foo/package_b":"0.1.0";
	"foo/package_b":"0.1.0" -> "foo/package_c":"0.1.0";
}

Reply:
Replaced `assertTrue` with `assertEqual` according to your suggestion.

The situation in the above comment due to `currentPackage().getResolution()` being executed in code generation/ modification step in `CompilerTask`. It invokes resolution engine multiple times with `--dump-graph` flag resulting in multiple graphs printing to the console.